### DISCUSSION TITLE
Global Javascript transition

### DISCUSSION THREAD

**Discussion by BenjaminOddou** [nuxt/nuxt#18834] [Questions]
As for now I didn't succeed to implement correctly nuxt transition with JS hooks in a "nuxt" way inside the `app.vue` component.

For now I have in my `app.vue` :

```vue
<script setup lang="ts">
const onBeforeLeave = (el) => {}
const onLeave = (el, done) => {}
const onEnter = (el, done) => {}
const onAfterEnter = (el) => {}
</script>

<template>
        <router-view v-slot="{ Component }">
          <Transition
            mode="out-in"
            :css="false"
            @before-leave="onBeforeLeave"
            @leave="onLeave"
            @enter="onEnter"
            @after-enter="onAfterEnter"
          >
            <component :is="Component" />
          </Transition>
        </router-view>
</template>

I would need to use something similar to [this](https://nuxt.com/docs/getting-started/transitions#javascript-hooks)

**Comment by BenjaminOddou** [comment]
After testing a bit, I think the correct syntax should be this as explained in the [docs](https://nuxt.com/docs/getting-started/transitions#transition-with-nuxtpage) for global transition but its generating lot of errors

`app.vue` :

```vue
<script setup lang="ts">
const onBeforeLeave = (el) => {}
const onLeave = (el, done) => {}
const onEnter = (el, done) => {}
const onAfterEnter = (el) => {}
</script>

<template>
  <div>
        <NuxtPage
          :transition="{
            mode: 'out-in',
            css: false,
            onBeforeLeave,
            onLeave,
            onEnter,
            onAfterEnter,
          }"
        />
  </div>
</template>
```

`errors` :
![errors](https://user-images.githubusercontent.com/85166574/217097621-debd0f14-c391-46a8-8849-426854b84253.png)

**Reply by danielroe** [reply]
Yes, that looks right to me, but I can't investigate as the warnings you're getting are coming from GSAP and I don't see any mention of that in the sample code you've given above.

**Reply by BenjaminOddou** [reply]
Hello @danielroe, I created a small repro of the problem [here](https://stackblitz.com/edit/nuxt-starter-bxqwv5?file=app.vue).

Note that when using the `Nuxt syntax` I have the following errors :
![Capture d’écran 2023-02-13 à 17 35 00](https://user-images.githubusercontent.com/85166574/218517356-d211aedb-d3a6-473a-b796-abbd13e20dde.png)

but using the `router-view syntax` is working as expected :
![Capture d’écran 2023-02-13 à 17 36 35](https://user-images.githubusercontent.com/85166574/218517515-24a63c0b-0128-4756-a6d4-159f5fb8fc19.png)

**Reply by BenjaminOddou** [reply]
Hello @danielroe , did you had time to look at this ? 😊
